src/main/java/co/cask/gcp/gcs/sink/GCSMultiBatchSink.java

```diff
@@ -0,0 +1,191 @@
+
+      Job job = JobUtils.createInstance();
+      Configuration outputConfig = job.getConfiguration();
+
+      outputConfig.set(FileOutputFormat.OUTDIR, config.getOutputDir(context.getLogicalStartTime(), name));
+      Map<String, String> gcsProperties = GCPUtils.getFileSystemProperties(config);
+      for (Map.Entry<String, String> entry : gcsProperties.entrySet()) {
+        outputConfig.set(entry.getKey(), entry.getValue());
+      }
+
+
+      outputConfig.set(RecordFilterOutputFormat.PASS_VALUE, name);
+      outputConfig.set(RecordFilterOutputFormat.ORIGINAL_SCHEMA, schema);
+      String format = config.getOutputFormat();
+      outputConfig.set(RecordFilterOutputFormat.FORMAT, format);
+
+
+      outputConfig.set(RecordFilterOutputFormat.FILTER_FIELD, config.splitField);
+      job.setOutputValueClass(org.apache.hadoop.io.NullWritable.class);
+      if (RecordFilterOutputFormat.AVRO.equals(format)) {
+        org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(schema);
+        Map<String, String> configuration =
+                FileSetUtil.getAvroCompressionConfiguration(config.codec, schema, false);
+        for (Map.Entry<String, String> entry : configuration.entrySet()) {
+          outputConfig.set(entry.getKey(), entry.getValue());
+        }
+        AvroJob.setOutputKeySchema(job, avroSchema);
+      } else if (RecordFilterOutputFormat.ORC.equals(format)) {
+        StringBuilder builder = new StringBuilder();
+        co.cask.hydrator.common.HiveSchemaConverter.appendType(builder, Schema.parseJson(schema));
+        outputConfig.set("orc.mapred.output.schema", builder.toString());
+      } else if (RecordFilterOutputFormat.PARQUET.equals(format)) {
+        Map<String, String> configuration =
+                FileSetUtil.getParquetCompressionConfiguration(config.codec, schema, false);
+        for (Map.Entry<String, String> entry : configuration.entrySet()) {
+          outputConfig.set(entry.getKey(), entry.getValue());
+        }
+      } else {
+        // Encode the delimiter to base64 to support control characters. Otherwise serializing it in Cconf would result
+        // in an error
+        outputConfig.set(RecordFilterOutputFormat.DELIMITER, config.getDelimiter());
+      }
+
+
+      context.addOutput(Output.of(config.getReferenceName(),
+              new SinkOutputFormatProvider(RecordFilterOutputFormat.class.getName(), outputConfig)));
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord input,
+                        Emitter<KeyValue<NullWritable, StructuredRecord>> emitter) {
+    emitter.emit(new KeyValue<>(org.apache.hadoop.io.NullWritable.get(), input));
+  }
+
+  @VisibleForTesting
+  GCSMultiBatchSinkConfig getConfig() {
+    return config;
+  }
+
+  /**
+   * Sink configuration.
+   */
+  public static class GCSMultiBatchSinkConfig extends GCSBatchSink.GCSBatchSinkConfig {
+
+    @Description("The prefix for the folder in GCS")
```

Comment:
What is this prefix for? It seems redundant, as it can just be included in the path.

Reply:
The prefix is for the destination directories since they are created automatically. I need to to pass the the db name so I'll have the context from which DB I have consumed the tables

Reply:
Right, but why can't the db name just be in the base path to begin with?